### PR TITLE
Update extension to support Juvix v0.6.3

### LINKED ISSUE #127: Adapt to the new CLI compile command syntax
Adapt to https://github.com/anoma/juvix/pull/2713

### LINKED ISSUE #130: Typecheck and compile functionality fails
## Summary

The `typecheck` and `compile` buttons in the Juvix VS Code plugin result in errors preventing correct code highlighting and definition search. On the CLI, the same type checking and compilation works without problems.

## Example Project

following example project

```
.
├── Hello.juvix
└── Package.juvix
```

`Package.juvix`
```juvix
module Package;

import PackageDescription.V2 open;

package : Package :=
  defaultPackage
    {name := "anoma";
     version := mkVersion 0 1 0;
     dependencies := [ defaultStdlib
                     ; github "anoma" "juvix-anoma-stdlib" "main"
                     ]};
```

`Hello.juvix`
```juvix
module Hello;
import Stdlib.Prelude open;
import Anoma.Transaction open;

main : String := "Hello world!";
```

After running `juvix dependencies update`, the following `juvix.lock.yaml` file is correctly produced
```yaml
version: 2
checksum: ba5a40db0be704d362c2a0184663856a1451c907e1b06d27e135c1eafb769209
dependencies:
- path: .juvix-build/stdlib/
  dependencies: []
- git:
    name: anoma_juvix-anoma-stdlib
    ref: 5c72a64d5c9d17ad289c3c72d340a7ef3857bd10
    url: https://github.com/anoma/juvix-anoma-stdlib
  dependencies:
  - path: .juvix-build/stdlib/
    dependencies: []
```

Afterward, the `juvix typecheck Hello.juvix` and `juvix compile native Hello.juvix`  command run successfully in the terminal.

## Problem

Triggering the typecheck through the vscode plugin button results in the folliwung error:

```sh
/Users/<USERNAME>/Projects/Anoma/sandbox/Package.juvix:1:1: error:
The dependency /private/var/folders/62/fl3kvblx6jj7p013xzpmdylc0000gn/T/.juvix-build5syPKX/stdlib/ is declared in the package's juvix.yaml but is not declared in the lockfile: /Users/<USERNAME>/Projects/Anoma/sandbox/juvix.lock.yaml
Try running the following command:
juvix dependencies update

 *  The terminal process "/bin/zsh '-l', '-c', '/Users/<USERNAME>/.local/bin/juvix --internal-build-dir /var/folders/62/fl3kvblx6jj7p013xzpmdylc0000gn/T/.juvix-build5syPKX  typecheck /Users/<USERNAME>/Projects/Anoma/sandbox/Hello.juvix'" failed to launch (exit code: 1). 
```

Triggering compilation through the vscode compilation button results in an error too:
```sh
Invalid option `--output'

Usage: juvix compile COMMAND

  Compile a Juvix file

 *  The terminal process "/bin/zsh '-l', '-c', '/Users/<USERNAME>/.local/bin/juvix --internal-build-dir /var/folders/62/fl3kvblx6jj7p013xzpmdylc0000gn/T/.juvix-buildslaN1t compile --output /var/folders/62/fl3kvblx6jj7p013xzpmdylc0000gn/T/.juvix-buildbNPvRt/out /Users/<USERNAME>/Projects/Anoma/sandbox/Hello.juvix && /var/folders/62/fl3kvblx6jj7p013xzpmdylc0000gn/T/.juvix-buildbNPvRt/out'" failed to launch (exit code: 1). 
```

## Info

**System**
arch: Apple M1
OS: macOS 14.5 (23F79)
Juvix: v0.6.2-371f8f2

**VS Code**
Version: 1.90.0 (Universal)
Commit: 89de5a8d4d6205e5b11647eb6a74844ca23d2573
Date: 2024-06-04T19:34:44.157Z (5 days ago)
Electron: 29.4.0
ElectronBuildId: 9593362
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.5.0

### PR DESCRIPTION
**Pull request by jonaprieto**
This PR makes the extension usable compared to the previous 0.5.X version plus some enchaments.

- Closes #127 
- Closes #130 
- Support for Juvix Markdown files, including syntax highlighting for Markdown parts.
- Better codelens plus Package.juvix template
- ...

Long story:
The extension was quite outdated, unfortunately, and needed several updates for the vscode and npm packages (now with Node v22 instead of v18). This required several changes in the package.json file and resolving some TypeScript issues to ensure that the extension's features such as type-checking and compilation worked as expected. However, there are still some open issues, such as type information. Other updates included modifying the README, removing outdated features like devTasks, and deleting some old commands. As part of the enhancements, this PR introduced support for Juvix markdown files, enabling this mode to have the same features as regular Juvix files. We also made some essential removals for easier maintenance, such as unicode-input support, which was difficult to maintain and had many bugs from the original lean3 input method. Instead, we now recommend installing czhang03.unicode-math-input input as part of the extension Juvix pack, which alleviates this issue. Additionally, the user configuration menu now includes some additional options for these tasks. We simplified the logic for codelens for inserting module declarations and added a Package template when opening empty Package.juvix files.

Testing is hard with these extensions, so some help is welcomed in this regard. 
One way to test this is to install the following .vsix file directly. See this guide https://code.visualstudio.com/docs/editor/extension-marketplace#_install-from-a-vsix

- [juvix-mode-0.2.0.vsix.zip](https://github.com/user-attachments/files/16120496/juvix-mode-0.2.0.vsix.zip)

Ps. NPM made me waste so much time. :( I hope to find something better.

### CODE REVIEW

**Comment by lukaszcz** [comment]
Why do we attribute copyright to Microsoft everywhere?

**Comment by lukaszcz** [comment]
We should add a LICENSE file

**Comment by jonaprieto** [comment]
> Why do we attribute copyright to Microsoft everywhere?

As far as I remember, it was needed to upload the extension to their marketplace.

**Comment by jonaprieto** [comment]
> We should add a LICENSE file

We have one in the root directory.

**Comment by lukaszcz** [comment]
> We should add a LICENSE file

Sorry, somehow didn't notice

**Comment by lukaszcz** [comment]
> > Why do we attribute copyright to Microsoft everywhere?
> 
> As far as I remember, it was needed to upload the extension to their marketplace.

> 
> That's weird. So basically it's a requirement to hand over the copyright to Microsoft if you want to publish the extension on their marketplace?
> 
> At least the license is GPL.

Agree. Maybe this is no longer or never was the case. But I remember needing to add this.
https://github.com/anoma/vscode-juvix/blob/26b76fbed104e3fde544ec8bf1000ff5b9b1fce2/.eslintrc.js#L20-L24

We could try to remove them, upload the code and see if it passes the control.

File: src/check.ts [hunk 1]
```diff
@@ -25,8 +25,8 @@ export async function activate(context: vscode.ExtensionContext) {
         const typecheckerCall = [
           config.getJuvixExec(),
           config.getGlobalFlags(),
-          '--only-errors',
```
Comment (lukaszcz) [inline src/check.ts:28]: Why was `--only-errors` deleted here? That's the reason we get all the "Compiling" messages

File: src/tasks.ts [hunk 2]
```diff
@@ -131,7 +131,25 @@ export class JuvixTaskProvider implements vscode.TaskProvider {
...
+        reveal: vscode.TaskRevealKind.Always,
+      },
+      {
+        command: 'update-dependencies',
+        args: [],
+        group: vscode.TaskGroup.Build,
+        reveal: vscode.TaskRevealKind.Always,
+      },
+      {
+        command: 'generate-isabelle',
```
Comment (lukaszcz) [inline src/tasks.ts]: There is no `generate-isabelle` command in the `juvix` binary. The command is called `isabelle`. This just fails anytime it's run.

Comment (lukaszcz) [reply src/tasks.ts]: Perhaps we should change the name of the command in `juvix` to `generate-isabelle` (more descriptive)

File: src/check.ts (hunk 1 above)
Comment (lukaszcz) [reply src/check.ts:28]: Okay, `--only-errors` also suppresses the "Well done, it type checks!" message. We should remove it for now.

File: src/tasks.ts (hunk 2 above)
Comment (lukaszcz) [reply src/tasks.ts]: I fixed this. The command name in `juvix` should be changed in the future.

**Review by lukaszcz — DISMISSED** [review]

**Pull request merged**